From 1cea6c4528d069a2c77ec42ef305545b9e9609b7 Mon Sep 17 00:00:00 2001
From: Matthias Clasen <mclasen@redhat.com>
Date: Mon, 27 Sep 2010 16:43:37 -0400
Subject: [PATCH] gtk: Fix the build with current GTK+

gdk_drawable_get_screen has been removed in GTK3. Instead, use
gdk_window_get_screen.
---
 src/canberra-gtk.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/src/canberra-gtk.c b/src/canberra-gtk.c
index fffcfef..34446f5 100644
--- a/src/canberra-gtk.c
+++ b/src/canberra-gtk.c
@@ -505,7 +505,11 @@ int ca_gtk_play_for_event(GdkEvent *e, uint32_t id, ...) {
                 goto fail;
 
         if (e->any.window)
+#if GTK_CHECK_VERSION (2, 90, 8)
+                s = gdk_window_get_screen(e->any.window);
+#else
                 s = gdk_drawable_get_screen(GDK_DRAWABLE(e->any.window));
+#endif
         else
                 s = gdk_screen_get_default();
 
-- 
1.7.3

